authorization/invitation/service/invitation_service.go

```diff
@@ -162,3 +199,84 @@ func (s *invitationServiceImpl) Issue(ctx context.Context, issuingUserId uuid.UU
 
 	return nil
 }
+
+type invitationNotification struct {
+	invitation *invitationrepo.Invitation
+	roles      []string
+}
+
+// processTeamInviteNotifications sends an e-mail notification to a user.
+func (s *invitationServiceImpl) processTeamInviteNotifications(ctx context.Context, team *account.Identity, inviterName string,
+	notifications []invitationNotification, witURL string) {
+	teamName := team.IdentityResource.Name
+
+	spaceName, err := lookupSpaceName(ctx, witURL, *team.IdentityResource.ParentResourceID)
+	if err != nil {
+		// What to do here?? just log an error and die?
```

Comment:
Let's not just log and die. Let's return an error all the way up to the controller - and then return an 50x/40x response code because without the email the user has not really been invited to the space at all.

This can occur only in an exceptional case. The user who was inviting should then retry inviting the users to the space. ( we will end up with a bunch of unsent invites in the table, but that's better than silently failing and not asking the user to retry )

Reply:
Since this is no longer running in a separate thread I now return an error